### ISSUE TITLE
option to disable logger and thread

### ISSUE THREAD

**Issue by ShuangLiu1992** [davisking/dlib#1906]
is it possible to add a cmake option to disable thread and logger in dlib?
compiling dlib with emscripten in debug mode leads to logger linking error, also threads are currently disabled on the web due to sharedarraybuffer being disabled because of spectre.

```
wasm-0d75152e-673:1 Uncaught RuntimeError: invalid index into function table
    at void dlib::mfp_kernel_1_base_class<4ul>::mp_impl_T<dlib::mfp_kernel_1_base_class<4ul>::mp_null<dlib::mfp_kernel_1_base_class<4ul>::dummy> >::safe_clone<21ul>(dlib::stack_based_memory_block<21ul>&) (wasm-function[673]:0x145fd)
    at dlib::mfp_kernel_1_base_class<4ul>::mfp_kernel_1_base_class() (wasm-function[563]:0xf501)
    at dlib::member_function_pointer<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > const&, dlib::log_level const&, unsigned long long const, char const*>::member_function_pointer() (wasm-function[558]:0xefea)
    at dlib::logger::global_data::output_hook_container::output_hook_container() (wasm-function[601]:0x10b71)
    at dlib::logger::global_data::global_data() (wasm-function[595]:0x1082f)
    at dlib::logger::get_global_data() (wasm-function[554]:0xee69)
    at dlib::logger::logger(std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > const&) (wasm-function[659]:0x13cc8)
    at __cxx_global_var_init.17.1 (wasm-function[569]:0xfa5c)
    at _GLOBAL__sub_I_logger_kernel_1.cpp (wasm-function[1252]:0x2bf84)
    at __wasm_call_ctors (wasm-function[44]:0x2293)

```

**Comment by davisking** [comment]
I would build only the .cpp files from dlib you need to use for your task then.  Don't bother with the dlib cmake files and build your own thing for your own needs.  You don't need to do anything exciting to build dlib.  I realize the dlib cmake files look complicated, but that's because they work around a bunch of stuff like, bugs in older cmakes, how to build and link with cuda, finding and using libraries like libpng on your system and testing that they aren't broken (this is more common than you would think).  

TLDR, you can build dlib by just building its .cpp files with a vanilla C++ compiler and it's fine.  You don't have to setup cuda stuff or link to anything.  If there are parts you don't want don't build those .cpp files.  Read the docs about building without cmake: http://dlib.net/compile.html
